### PR TITLE
Move the StartScriptCommandBuilders to the Tentacle.Contracts to allow consumers of TentacleClient to use the builders.

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

[sc-66868]

As part of upgrading DynamicWorkers to use TentacleClient, a set of classes were copied from Server / Tentacle into DynamicWorkers. 

One of the areas was the StartScriptCommandBuilders. The builders provide a convenient way of creating the start script commands.

# Results

This PR exposes a set of StartScriptCommand builders in the Tentacler.Contracts library.
The builders were moved from the testing project and refactored to have better sensible defaults for consumers.

The idea would be that these builders can be used by DyanmicWorkers and also Server as an option refactor.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.CommonTestUtils/Builders/LatestStartScriptCommandBuilder.cs [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+﻿using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.Builders;
+
+namespace Octopus.Tentacle.CommonTestUtils.Builders
 {
     public class LatestStartScriptCommandBuilder : StartScriptCommandV3AlphaBuilder
     {
+        public LatestStartScriptCommandBuilder()
+        {
+            WithIsolation(ScriptIsolationLevel.NoIsolation);
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle.CommonTestUtils/Builders/LatestStartScriptCommandBuilder.cs:10]: This ensures there is no change to the StartScriptComamnd being used by the Tests

File: source/Octopus.Tentacle.Contracts/Builders/StartScriptCommandV3AlphaBuilder.cs [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
-        readonly List<ScriptFile> files = new List<ScriptFile>();
-        readonly List<string> arguments = new List<string>();
-        readonly Dictionary<ScriptType, string> additionalScripts = new Dictionary<ScriptType, string>();
-        StringBuilder scriptBody = new StringBuilder(string.Empty);
-        ScriptIsolationLevel isolation = ScriptIsolationLevel.NoIsolation;
+        readonly List<ScriptFile> files = new();
+        readonly List<string> arguments = new();
+        readonly Dictionary<ScriptType, string> additionalScripts = new();
+        StringBuilder scriptBody = new(string.Empty);
+        ScriptIsolationLevel isolation = ScriptIsolationLevel.FullIsolation;
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle.Contracts/Builders/StartScriptCommandV3AlphaBuilder.cs:14]: These defaults align with the usage in Server

File: source/Octopus.Tentacle.Tests/Integration/ScriptServiceV2Fixture.cs [hunk 3]
```diff
@@ -52,6 +53,7 @@ public async Task ShouldExecuteAScriptSuccessfully()
 
             var startScriptCommand = new StartScriptCommandV2Builder()
                 .WithScriptBodyForCurrentOs(windowsScript, bashScript)
+                .WithIsolation(ScriptIsolationLevel.NoIsolation)
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle.Tests/Integration/ScriptServiceV2Fixture.cs:56]: This ensures there is no behaviour changes for these tests

File: source/Octopus.Tentacle.Tests/Contracts/UniqueScriptTicketBuilderFixture.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+using System.Collections.Generic;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using NUnit.Framework;
+using Octopus.Tentacle.Contracts.Builders;
+
+namespace Octopus.Tentacle.Tests.Contracts
+{
+    [TestFixture]
+    public class UniqueScriptTicketBuilderFixture
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle.Tests/Contracts/UniqueScriptTicketBuilderFixture.cs:11]: Taken from Server. A follow-up PR could remove this from Server

File: source/Octopus.Tentacle.Contracts/Builders/UniqueScriptTicketBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Linq;
+
+namespace Octopus.Tentacle.Contracts.Builders
+{
+    public class UniqueScriptTicketBuilder
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle.Contracts/Builders/UniqueScriptTicketBuilder.cs:6]: Taken from Server. A follow-up PR could remove this from Server

**Review by LukeButters — APPROVED** [review]

**Pull request merged**